IPython/core/completer.py

```diff
@@ -1879,6 +1891,14 @@ def _complete(self, *, cursor_line, cursor_pos, line_buffer=None, text=None,
         self.line_buffer = line_buffer
         self.text_until_cursor = self.line_buffer[:cursor_pos]
 
+        # Do magic arg matches
+        for matcher in self.magic_arg_matchers:
+            matches = [(m, matcher.__qualname__) for m in matcher(line_buffer)]
```

Comment:
You actually get two of these: 
```
self.magic_arg_matchers = [
            self.magic_config_matches,
            self.magic_color_matches,
        ]
```

Origin is not super important but useful for debug, and to deduplicate completions in a smart ways. It's cheap enough o keep.